src/client/mod.rs

```diff
@@ -134,9 +136,19 @@ impl Client {
         Ok(Self { inner })
     }
 
-    pub(crate) fn emit_command_event(&self, emit: impl FnOnce(&Arc<dyn CommandEventHandler>)) {
+    pub(crate) fn emit_command_event(&self, generate_event: impl Fn() -> CommandEvent) {
```

Comment:
I spent a while trying to figure out a nice way to keep this as a `FnOnce` (which would save the `clone()`s I had to add in the operation executor code that emits events), but I had a hard time figuring out how to only clone the resulting event if it was actually being passed to `handle_command_event` twice (the code looked like [this](https://github.com/kmahar/mongo-rust-driver/blob/4d5b402ab8db969a2953dc0cb8932b880d8d5db8/src/client/mod.rs#L139-L162)), short of branching on whether one or both of monitoring and tracing were enabled and having near-duplicate code under each branch to create/unwrap the appropriate handler(s) and pass the event/event clone to them. 

in the end I got to thinking that the `clone()`s I had to add were likely pretty cheap, and that it seems rare someone would actually use both monitoring and tracing given they provide really similar functionality, so that paying the penalty of re-generating the event in those cases is probably fine. 

that said, if there is a nicer way to do what I was trying to do I'm not opposed to reworking this. thoughts?